### DISCUSSION TITLE
Cannot use any Torch modules

### DISCUSSION THREAD

**Discussion by nfurth1** [community/community#53880] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

When ever I try to use any Torch module I get an error:
import torch.multiprocessing as mp
ModuleNotFoundError: No module named 'torch.multiprocessing'
It does the same for any module I try to call

Cuda 11.1 is installed, using Windows 10
Torch was detecting gpus before now I cannot call torch.device to make sure

`
- aiosignal=1.3.1=pypi_0
- alabaster=0.7.13=pyhd8ed1ab_0
- arrow=1.2.3=pyhd8ed1ab_0
- astroid=2.12.11=py37h03978a9_0
- atomicwrites=1.4.1=pyhd8ed1ab_0
- attrs=23.1.0=pypi_0
- autopep8=1.6.0=pyhd8ed1ab_1
- babel=2.12.1=pyhd8ed1ab_1
- backcall=0.2.0=pyh9f0ad1d_0
- backports=1.0=pyhd8ed1ab_3
- backports.functools_lru_cache=1.6.4=pyhd8ed1ab_0
- bcrypt=3.2.2=py37hcc03f2d_0
- beautifulsoup4=4.12.2=pyha770c72_0
- binaryornot=0.4.4=py_1
- black=22.10.0=py37h03978a9_1
- blas=1.0=mkl
- bleach=6.0.0=pyhd8ed1ab_0
- brotlipy=0.7.0=py37hcc03f2d_1004
- bzip2=1.0.8=h8ffe710_4
- ca-certificates=2022.12.7=h5b45459_0
- certifi=2022.12.7=pyhd8ed1ab_0
- cffi=1.15.1=py37ha95fbe2_1
- chardet=5.0.0=py37h03978a9_0
- charset-normalizer=3.1.0=pyhd8ed1ab_0
- click=8.1.3=py37h03978a9_0
- cloudpickle=2.2.1=pyhd8ed1ab_0
- colorama=0.4.6=pyhd8ed1ab_0
- cookiecutter=2.1.1=pyh6c4a22f_0
- cryptography=38.0.2=py37h953a470_1
- cudatoolkit=11.1.1=heb2d755_10
- debugpy=1.6.3=py37hf2a7229_0
- decorator=5.1.1=pyhd8ed1ab_0
- defusedxml=0.7.1=pyhd8ed1ab_0
- diff-match-patch=20200713=pyh9f0ad1d_0
- dill=0.3.6=pyhd8ed1ab_1
- distlib=0.3.6=pypi_0
- docutils=0.19=py37h03978a9_0
- entrypoints=0.4=pyhd8ed1ab_0
- filelock=3.12.0=pypi_0
- flake8=4.0.1=pyhd8ed1ab_2
- frozenlist=1.3.3=pypi_0
- gettext=0.21.1=h5728263_0
- glib=2.74.0=h12be248_0
- glib-tools=2.74.0=h12be248_0
- googledrivedownloader=0.4=pypi_0
- grpcio=1.43.0=pypi_0
- gst-plugins-base=1.18.5=he07aa86_3
- gstreamer=1.18.5=hdff456e_3
- icu=58.2=ha925a31_3
- idna=3.4=pyhd8ed1ab_0
- imagesize=1.4.1=pyhd8ed1ab_0
- importlib-metadata=4.13.0=pypi_0
- importlib_metadata=4.11.4=hd8ed1ab_0
- importlib_resources=5.12.0=pyhd8ed1ab_0
- inflection=0.5.1=pyh9f0ad1d_0
- intel-openmp=2023.1.0=h57928b3_46319
- intervaltree=3.0.2=py_0
- ipykernel=6.16.2=pyh025b116_0
- ipython=7.33.0=py37h03978a9_0
- ipython_genutils=0.2.0=py_1
- isodate=0.6.1=pypi_0
- isort=5.11.5=pyhd8ed1ab_0
- jaraco.classes=3.2.3=pyhd8ed1ab_0
- jedi=0.18.2=pyhd8ed1ab_0
- jellyfish=0.9.0=py37hcc03f2d_1
- jinja2=3.1.2=pyhd8ed1ab_1
- jinja2-time=0.2.0=pyhd8ed1ab_3
- joblib=1.2.0=pypi_0
- jpeg=9e=hcfcfb64_3
- jsonschema=4.17.3=pyhd8ed1ab_0
- jupyter_client=7.4.9=pyhd8ed1ab_0
- jupyter_core=4.11.1=py37h03978a9_0
- jupyterlab_pygments=0.2.2=pyhd8ed1ab_0
- keyring=23.9.3=py37h03978a9_0
- lazy-object-proxy=1.7.1=py37hcc03f2d_1
- libclang=12.0.1=default_h81446c8_4
- libffi=3.4.2=h8ffe710_5
- libglib=2.74.0=h79619a9_0
- libiconv=1.17=h8ffe710_0
- libogg=1.3.4=h8ffe710_1
- libpng=1.6.38=h19919ed_0
- libsodium=1.0.18=h8d14728_1
- libspatialindex=1.9.3=h39d44d4_4
- libsqlite=3.39.2=h8ffe710_1
- libvorbis=1.3.7=h0e60522_0
- libzlib=1.2.12=h8ffe710_2
- markupsafe=2.1.2=pypi_0
- matplotlib-inline=0.1.6=pyhd8ed1ab_0
- mccabe=0.6.1=py_1
- mistune=2.0.5=pyhd8ed1ab_0
- mkl=2021.4.0=h0e2418a_729
- mkl-service=2.4.0=py37h75fcce0_0
- more-itertools=9.1.0=pyhd8ed1ab_0
- msgpack=1.0.5=pypi_0
- mypy_extensions=1.0.0=pyha770c72_0
- nbclient=0.7.0=pyhd8ed1ab_0
- nbconvert=7.3.1=pyhd8ed1ab_0
- nbconvert-core=7.3.1=pyhd8ed1ab_0
- nbconvert-pandoc=7.3.1=pyhd8ed1ab_0
- nbformat=5.8.0=pyhd8ed1ab_0
- nest-asyncio=1.5.6=pyhd8ed1ab_0
- networkx=2.6.3=pypi_0
- numpy-base=1.21.5=py37hca35cd5_3
- numpydoc=1.5.0=pyhd8ed1ab_0
- openssl=3.1.0=hcfcfb64_2
- packaging=23.1=pyhd8ed1ab_0
- pandas=1.3.5=pypi_0
- pandoc=2.19.2=h57928b3_2
- pandocfilters=1.5.0=pyhd8ed1ab_0
- paramiko=3.1.0=pyhd8ed1ab_0
- parso=0.8.3=pyhd8ed1ab_0
- pathspec=0.11.1=pyhd8ed1ab_0
- pcre2=10.37=hdfff0fc_1
- pexpect=4.8.0=pyh1a96a4e_2
- pickleshare=0.7.5=py_1003
- pillow=9.5.0=pypi_0
- pip=23.1.2=pypi_0
- pkgutil-resolve-name=1.3.10=pyhd8ed1ab_0
- platformdirs=3.4.0=pyhd8ed1ab_0
- pluggy=1.0.0=py37h03978a9_3
- prompt-toolkit=3.0.38=pyha770c72_0
- protobuf=3.20.3=pypi_0
- psutil=5.9.3=py37h51bd9d9_0
- ptyprocess=0.7.0=pyhd3deb0d_0
- pycodestyle=2.8.0=pyhd8ed1ab_0
- pycparser=2.21=pyhd8ed1ab_0
- pydocstyle=6.2.0=pyhd8ed1ab_0
- pyflakes=2.4.0=pyhd8ed1ab_0
- pygments=2.15.1=pyhd8ed1ab_0
- pylint=2.15.4=pyhd8ed1ab_0
- pyls-spyder=0.4.0=pyhd8ed1ab_0
- pynacl=1.5.0=py37ha54c9ec_1
- pyopenssl=23.1.1=pyhd8ed1ab_0
- pyqt=5.15.4=py37hf2a7229_0
- pyqt5-sip=12.9.0=py37hf2a7229_0
- pyrsistent=0.19.3=pypi_0
- pysocks=1.7.1=py37h03978a9_5
- python=3.7.12=h900ac77_100_cpython
- python-dateutil=2.8.2=pyhd8ed1ab_0
- python-fastjsonschema=2.16.3=pyhd8ed1ab_0
- python-lsp-black=1.1.0=pyhd8ed1ab_0
- python-lsp-jsonrpc=1.0.0=pyhd8ed1ab_0
- python-lsp-server=1.3.3=pyhd8ed1ab_0
- python-slugify=8.0.1=pyhd8ed1ab_0
- python_abi=3.7=3_cp37m
- pytoolconfig=1.2.5=pyhd8ed1ab_0
- pytz=2023.3=pyhd8ed1ab_0
- pywin32=303=py37hcc03f2d_0
- pywin32-ctypes=0.2.0=py37h03978a9_1005
- pyyaml=6.0=py37hcc03f2d_4
- pyzmq=24.0.1=py37h7347f05_0
- qdarkstyle=3.0.2=pyhd8ed1ab_0
- qstylizer=0.2.2=pyhd8ed1ab_0
- qt-main=5.15.2=he8e5bd7_7
- qtawesome=1.2.3=pyhd8ed1ab_0
- qtconsole=5.2.2=pyhd8ed1ab_0
- qtpy=2.3.1=pyhd8ed1ab_0
- ray=2.0.1=pypi_0
- rdflib=6.3.2=pypi_0
- requests=2.29.0=pyhd8ed1ab_0
- rope=1.7.0=pyhd8ed1ab_0
- rtree=1.0.1=py37h13cc57e_0
- ruamel.yaml=0.17.21=py37hcc03f2d_1
- ruamel.yaml.clib=0.2.6=py37hcc03f2d_1
- setuptools=67.7.2=pyhd8ed1ab_0
- sip=6.5.1=py37hf2a7229_2
- six=1.16.0=pyh6c4a22f_0
- snowballstemmer=2.2.0=pyhd8ed1ab_0
- sortedcontainers=2.4.0=pyhd8ed1ab_0
- soupsieve=2.3.2.post1=pyhd8ed1ab_0
- sphinx=5.3.0=pyhd8ed1ab_0
- sphinxcontrib-applehelp=1.0.4=pyhd8ed1ab_0
- sphinxcontrib-devhelp=1.0.2=py_0
- sphinxcontrib-htmlhelp=2.0.1=pyhd8ed1ab_0
- sphinxcontrib-jsmath=1.0.1=py_0
- sphinxcontrib-qthelp=1.0.3=py_0
- sphinxcontrib-serializinghtml=1.1.5=pyhd8ed1ab_2
- spyder=5.2.2=py37h03978a9_3
- spyder-kernels=2.2.1=py37h03978a9_1
- sqlite=3.39.2=h8ffe710_1
- tabulate=0.9.0=pypi_0
- tbb=2021.7.0=h91493d7_0
- tensorboardx=2.6=pypi_0
- text-unidecode=1.3=py_0
- textdistance=4.5.0=pyhd8ed1ab_0
- threadpoolctl=3.1.0=pypi_0
- three-merge=0.1.1=pyh9f0ad1d_0
- tinycss2=1.2.1=pyhd8ed1ab_0
- toml=0.10.2=pyhd8ed1ab_0
- tomli=2.0.1=pyhd8ed1ab_0
- tomlkit=0.11.7=pyha770c72_0
- torch=1.9.0+cu111=pypi_0
- torch-cluster=1.5.9=pypi_0
- torch-geometric=2.0.3=pypi_0
- torch-scatter=2.0.9=pypi_0
- torch-sparse=0.6.12=pypi_0
- torch-spline-conv=1.2.1=pypi_0
- torchaudio=0.9.0=pypi_0
- torchvision=0.10.0+cu111=pypi_0
- tornado=6.2=py37hcc03f2d_0
- tqdm=4.65.0=pypi_0
- traitlets=5.9.0=pyhd8ed1ab_0
- typed-ast=1.5.4=py37hcc03f2d_0
- typing=3.10.0.0=pyhd8ed1ab_0
- typing-extensions=4.5.0=hd8ed1ab_0
- typing_extensions=4.5.0=pyha770c72_0
- ucrt=10.0.22621.0=h57928b3_0
- ujson=5.5.0=py37h7f67f24_0
- unidecode=1.3.6=pyhd8ed1ab_0
- urllib3=1.26.15=pyhd8ed1ab_0
- vc=14.3=h3d8a991_11
- virtualenv=20.22.0=pypi_0
- vs2015_runtime=14.34.31931=h4c5c07a_11
- watchdog=2.1.9=py37h03978a9_0
- wcwidth=0.2.6=pyhd8ed1ab_0
- webencodings=0.5.1=py_1
- wheel=0.40.0=pyhd8ed1ab_0
- win_inet_pton=1.1.0=pyhd8ed1ab_6
- wrapt=1.14.1=py37hcc03f2d_0
- yacs=0.1.8=pypi_0
- yaml=0.2.5=h8ffe710_2
- yapf=0.33.0=pyhd8ed1ab_0
- zeromq=4.3.4=h0e60522_1
- zipp=3.15.0=pyhd8ed1ab_0
- zlib=1.2.12=h8ffe710_2
- zstd=1.5.2=h7755175_4

`

**Comment by bbkx226** [comment]
The error you're seeing when trying to import modules from PyTorch could be because the PyTorch library version may not be compatible with other libraries you're using.

You mentioned using CUDA 11.1 and Windows 10, so it's possible that you need to install a specific version of PyTorch that's compatible with your setup.

Here's what you can try: First, check the version of PyTorch you have installed by running ```print(torch.__version__)``` in a Python console. If the version you have isn't compatible with your CUDA installation or other dependencies, you can try installing a different version of PyTorch using the command ```pip install torch==<version_number>``` where <version_number> is the version you want to install.

If this doesn't work, you may need to update or install other dependencies such as NumPy that PyTorch requires. You can do this by running ```pip install <library_name>``` for each library you need to install or update.

It's also a good idea to check your Python environment to ensure that you're using the correct version of Python and that your PATH is set up correctly.

**Comment by ncerutti** [comment]
Given that you are using Windows, it might be a matter of path: if you run the command `$env:CUDA_PATH` what is the result? Does it link to the right folder and right version?

Also, does it fail to find whatever module you import? For example, what is the output of this?
```
import torch
print(“Pytorch CUDA Version is “, torch.version.cuda)
```

In any case, I would perhaps trying with another version of torch (and CUDA, if needed). Also, if it is installed, remember to disable GeForce Experience if/when you reinstall CUDA.
